lib/surgex/parser/parsers/boolean_parser.ex

```diff
@@ -1,10 +1,7 @@
 defmodule Surgex.Parser.BooleanParser do
   @moduledoc false
 
-  @spec call(nil) :: {:ok, nil}
-  @spec call(true) :: {:ok, true}
-  @spec call(false) :: {:ok, false}
-  @spec call(String.t()) :: {:ok, true | false} | {:error, :invalid_boolean}
+  @spec call(Surgex.Types.json_value()) :: {:ok, true | false} | {:error, :invalid_boolean}
   def call(nil), do: {:ok, nil}
   def call("0"), do: {:ok, false}
   def call("1"), do: {:ok, true}
```

Comment:
I'm wondering if we should support such implicit coercion. Maybe that should be controlled with an explicit option. But that would break backward compatibility, OTOH.
A bit off-topic.

Reply:
@hot-cheese Yes, this is an interesting topic. I'm currently leaning towards what's described [here](https://www.tomdalling.com/blog/software-design/thoughts-on-schema-library-design/) under "Coercion Is Context-Dependent" section. 

In short: if it's a body of POST request which is a JSON, we probably should not coerce at all. However if it's a param in GET request, it has to be a string, because you can only send strings (and arrays) in URL params. Problem is that Surgex has no way to know which case it is. But the controller where it's used knows that. So maybe indeed it would be nice to have it parameterized there and based on this we could have a coercion or not. But this is definitely out of scope of this PR 😉